src/Factories/TestCaseMethodFactory.php

```diff
@@ -203,4 +208,14 @@ public static function $dataProviderName()
 
         EOF;
     }
+
+    /**
+     * Execute the 'after' callback, if set.
+     */
+    public function __destruct()
```

Comment:
In my experience, putting logic in destructor is a dangerous practice. Even though it may work now, it relies on assumptions about the timing of when the object will be destructed that can (accidentally) change in the future. For instance, the testing framework may keep objects in memory for longer, adding any circular dependency would make this behavior subject to the garbage collection, etc.

It would be more reliable (and less magical) to make the invocation explicit by calling a method.